### PR TITLE
refactor: move common platform code projects to JS Toolkit 

### PR DESCRIPTION
**Pull request by izaera**
This moves base target platforms projects to js-toolkit workspace and makes target platforms as lean as possible.

This has the following benefits:

1. The js-toolkit workspace has TypeScript support
2. The js-toolkit workspace is integrated in the general yarn workspace, allowing to access related projects through yarn link, for example
3. The target-platforms projects is only used as a lean repository of bundler configuration files/marker projects, not as a real development project with lots of source code. This is good because target-platforms is detached from the general yarn workspace (to avoid polluting dependency tree) and has a very limited build support.

I've tested that nothing broke with our fantabulous JS Toolkit QA scripts :tada:

### CODE REVIEW

**Comment by izaera** [comment]
Note that, once we merge this, we are ready to do the product review.

However, I'll send a PR with the scripts for local testing with Verdaccio first, so that I can then explain how can this be tested.

File: projects/js-toolkit/packages/portal-adapt-base/package.json [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
 	"dependencies": {
 		"@liferay/portal-base": "^1.0.0",
 		"cross-spawn": "^7.0.0",
 		"escape-string-regexp": "^2.0.0",
 		"globby": "^10.0.1"
 	},
 	"description": "Base library for adaptation autopresets",
-	"main": "./src/index.js",
+	"license": "LGPL-3.0",
+	"main": "lib/index.js",
```
Comment (bryceosterhaus) [inline projects/js-toolkit/packages/portal-adapt-base/package.json:11]: Just to verify, we will need to re-build this during development anytime we make changes?

File: projects/js-toolkit/tsconfig.json [hunk 2]
```diff
@@ -2,13 +2,13 @@
 	"compileOnSave": true,
 	"compilerOptions": {
 		"esModuleInterop": true,
-		"lib": ["ES2015", "DOM"],
+		"lib": ["ES2015"],
```
Comment (bryceosterhaus) [inline projects/js-toolkit/tsconfig.json:5]: This might be for a different thread, but any thoughts on upgrading to es2021?

Comment (izaera) [reply projects/js-toolkit/tsconfig.json:5]: Apparently ES2021 is only [fully supported in node v16](https://stackoverflow.com/questions/67371787/what-typescript-configuration-produces-output-closest-to-node-js-16-capabilities), so we may want to defer this, as we usually say that we use node LTS version as reference (which [is now 14](https://nodejs.org/en/)) in all our tooling...

**Pull request merged**